### PR TITLE
chore: Standardization and cleanup

### PR DESCRIPTION
**Pull request by deluca-mike**
# Description

# Integrations Checklist

- [ ] Have any function signatures changed? If yes, outline below.
- [ ] Have any features changed or been added? If yes, outline below.
- [ ] Have any events changed or been added? If yes, outline below.
- [ ] Has all documentation been updated?

# Changelog

### CODE REVIEW

File: contracts/ERC20.sol [hunk 1]
```diff
@@ -63,14 +63,14 @@ contract ERC20 is IERC20 {
...
-        require(deadline >= block.timestamp, "ERC20:P:EXPIRED");
+    function permit(address owner_, address spender_, uint256 amount_, uint256 deadline_, uint8 v_, bytes32 r_, bytes32 s_) external override {
+        require(deadline_ >= block.timestamp, "ERC20:P:EXPIRED");
 
         // Appendix F in the Ethereum Yellow paper (https://ethereum.github.io/yellowpaper/paper.pdf), defines
         // the valid range for s in (301): 0 < s < secp256k1n ÷ 2 + 1, and for v in (302): v ∈ {27, 28}.
         require(
-            uint256(s) <= 0x7FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF5D576E7357A4501DDFE92F46681B20A0 &&
-            (v == 27 || v == 28),
+            uint256(s_) <= 0x7FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF5D576E7357A4501DDFE92F46681B20A0 &&
```
Comment (vbidin) [inline contracts/ERC20.sol]: You can also mark the limit explicitly as a `uint256`.

File: contracts/interfaces/IERC20.sol [hunk 2]
```diff
@@ -133,9 +133,9 @@ interface IERC20 {
 
     /**
      *  @dev    Returns the permit type hash.
-     *  @return hash_ The permit type hash.
+     *  @return permitTypeHash_ The permit type hash.
```
Comment (vbidin) [inline contracts/interfaces/IERC20.sol]: Hmm, should this be `permitTypehash` instead? Based on the name of the function.

File: contracts/test/ERC20.t.sol [hunk 3]
```diff
@@ -10,240 +10,243 @@ import { MockERC20 } from "./mocks/MockERC20.sol";
 
 contract ERC20BaseTest is TestUtils {
 
-    bytes constant ARITHMETIC_ERROR = abi.encodeWithSignature("Panic(uint256)", 0x11);
+    address internal immutable self = address(this);
```
Comment (vbidin) [inline contracts/test/ERC20.t.sol:15]: AFAIK immutable should be all caps as well.

File: contracts/test/ERC20.t.sol [hunk 4]
```diff
@@ -10,240 +10,243 @@ import { MockERC20 } from "./mocks/MockERC20.sol";
...
 
-    MockERC20 token;
+    bytes internal constant ARITHMETIC_ERROR = abi.encodeWithSignature("Panic(uint256)", 0x11);
 
-    address internal immutable self = address(this);
+    MockERC20 internal _token;
 
-    function setUp() public virtual {
-        token = new MockERC20("Token", "TKN", 18);
+    function setUp() external virtual {
```
Comment (vbidin) [inline contracts/test/ERC20.t.sol:21]: The style guide has the test functions marked as public, but not sure what the convention is.

File: contracts/test/ERC20.t.sol [hunk 5]
```diff
@@ -303,58 +310,56 @@ contract ERC20PermitTest is TestUtils {
...
             )
         );
     }
 
     // Returns a valid `permit` signature signed by this contract's `owner` address
-    function _getValidPermitSignature(uint256 value_, address owner_, uint256 ownerSk_, uint256 deadline_) internal returns (uint8 v_, bytes32 r_, bytes32 s_) {
-        bytes32 digest = _getDigest(owner_, spender, value_, nonce, deadline_);
-        ( uint8 v, bytes32 r, bytes32 s ) = vm.sign(ownerSk_, digest);
-        return (v, r, s);
+    function _getValidPermitSignature(uint256 amount_, address owner_, uint256 ownerSk_, uint256 deadline_) internal returns (uint8 v_, bytes32 r_, bytes32 s_) {
```
Comment (vbidin) [inline contracts/test/ERC20.t.sol]: Maybe reorder parameters based on order in the function call?

File: contracts/test/ERC20.t.sol (hunk 3 above)
Comment (deluca-mike) [reply contracts/test/ERC20.t.sol:15]: You'd think that, but check out issue 9554 in solidity's issues... They don't consider immutables constant.

File: contracts/test/ERC20.t.sol (hunk 4 above)
Comment (deluca-mike) [reply contracts/test/ERC20.t.sol:21]: ight be just a copy and paste where it wasn't considered.

File: contracts/test/ERC20.t.sol (hunk 5 above)
Comment (deluca-mike) [reply contracts/test/ERC20.t.sol]: Yup, but I'll do one slightly better.

**Pull request merged**